src/renate/updaters/experimental/er.py

```diff
@@ -219,10 +219,11 @@ def _update_memory_buffer(self, step_output: STEP_OUTPUT) -> None:
 
     def _set_memory_loader(self) -> None:
         """Create a memory loader from a memory buffer."""
-        if self._memory_loader is None and len(self._memory_buffer) >= self._memory_batch_size:
+        memory_batch_size = int(self._batch_memory_frac * self._batch_size)
+        if self._memory_loader is None and len(self._memory_buffer) >= memory_batch_size:
```

Comment:
Because this is creating the memory loader, so we want to make sure it will return at least one batch. In `_sample_memory_buffer` it should then suffice to check for the existence of a memory loader, so I removed the superfluous second part.